### ISSUE TITLE
multiline_parameters - doesn't pick up nested function calls

### ISSUE THREAD

**Issue by allen-zeng** [realm/SwiftLint#1923]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Bug Report
When there is a nested function call to as the parameter of another function call, the `multiline_parameters` rule doesn't treat the nested call as a parameter on its own:
```swift
callingFunc1(callingFunc2(p1, p2: p2),
    p3: p3, p4: p4)
```

The expected correct structure should be either one of the following:
```swift
callingFunc1(
    callingFunc2(p1, p2: p2),
    p3: p3,
    p4: p4)

callingFunc1(callingFunc2(p1, p2: p2), p3: p3, p4: p4)
```

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ echo "callingFunc1(callingFunc2(\np1,\np2: p2),\np3: p3)" | swiftlint --use-stdin 
Done linting! Found 0 violations, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? **0.23.1**
* Installation method used (Homebrew, CocoaPods, building from source, etc)? **Homebrew or CocoaPods**, happens on both
* Paste your configuration file:

```yml
disabled_rules:
  - closing_brace
  - cyclomatic_complexity
  - force_cast
  - force_try
  - identifier_name
  - line_length
  - nesting
  - notification_center_detachment
  - todo
  - trailing_comma
  - trailing_whitespace
  - type_name
  - weak_delegate
  - file_length
excluded:
  - Carthage
  - Pods
colon:
  flexible_right_spacing: true
line_length: 400
type_body_length:
  - 2048
  - 2048
function_body_length:
  - 400
  - 400
force_try:
  severity: warning
reporter: "xcode"
large_tuple:
  - 4
  - 4
opt_in_rules:
  - closure_end_indentation
  - closure_spacing
  - empty_count
  - explicit_init
  - fatal_error_message
  - first_where
  - joined_default_parameter
  - multiline_parameters
  - overridden_super_call
  - prohibited_super_call
  - vertical_parameter_alignment_on_call
```

* Which Xcode version are you using (check `xcode-select -p`)? **9.0**

**Comment by ornithocoder** [comment]
Hi @allen-zeng, did you mean `multiline_parameters` or `multiline_arguments`? The first one is for function and method definitions and the latter for calls.

**Comment by ornithocoder** [comment]
```swift
// foo shouldn't trigger. All parameters are in separate lines
func foo(param1: String,
         param2: String,
         param3: String) -> String { }

// bar shouldn't trigger. All parameters are in separate lines
func bar(param1: String,
         param2: String,
         param3: Int) -> Int { }

// `example1` shouldn't trigger because parameters and argument calls are in separate lines
func example1(param1: String,
              param2: String,
              param3: Int = bar(param1: param1,
                                param2: p2,
                                param3: param3)) -> Int { }

// `example2` triggers both rules, for parameters and arguments
func example2(param1: String,
              param2: String, param3: Int = bar(param1: param1, param2: p2,
                                                param3: param3)) -> Int { }

// `example3` triggers for the nested arguments passed as param3
example3(param1: String,
         param2: String,
         param3: bar(param1: param1, param2: p2,
                     param3: param3)) -> Int { }
```

```
$ swiftlint --enable-all-rules
Linting Swift files in current working directory
Linting 'file.swift' (1/1)
file.swift:20:65: warning: Multiline Arguments Violation: Arguments should be either on the same line, or one per line. (multiline_arguments)
file.swift:26:38: warning: Multiline Arguments Violation: Arguments should be either on the same line, or one per line. (multiline_arguments)
file.swift:19:6: warning: Multiline Parameters Violation: Functions and methods parameters should be either on the same line, or one per line. (multiline_parameters)
Done linting! Found 3 violations, 0 serious in 1 file.
```

**Comment by ornithocoder** [comment]
With the example you provided:

```
$ cat file.swift 
callingFunc1(callingFunc2(p1, p2: p2),
             p3: p3, p4: p4)
```

```
$ swiftlint --enable-all-rules
Linting Swift files in current working directory
Linting 'file.swift' (1/1)
file.swift:2:22: warning: Multiline Arguments Violation: Arguments should be either on the same line, or one per line. (multiline_arguments)
Done linting! Found 1 violation, 0 serious in 1 file.
```

**Comment by allen-zeng** [comment]
Ah I see, seems to be that this is already covered, thanks for clarifying @ornithocoder !

**Comment by ornithocoder** [comment]
My pleasure, @allen-zeng!
